### ISSUE TITLE
FieldArray's unshift mutator does not update state correctly

### ISSUE THREAD

**Issue by rrjohnson85** [final-form/react-final-form-arrays#36]
### Are you submitting a **bug report** or a **feature request**?
Bug

### What is the current behavior?
When using the `unshift` mutator to add an item to the beginning of an array, the touched state from the item previously at index zero (if one exists) is maintained.

### What is the expected behavior?
I would expect the item that was added to the beginning of the array to have a touched state of false.

### Sandbox Link
N/A, but I will happily provide one if needed.

### What's your environment?

- final-form: 4.8.3
- react-final-form: 3.6.4
- final-forms-array: 1.0.4
- react-final-forms-array: 1.0.6
- React: 16.4.1
- Chrome: 68.0.3440.75 

### Other information
N/A

**Comment by tatchi** [comment]
Hello,

If it can for further investigation, I think it might be related to #33

**Comment by 311chaos** [comment]
I have a similar issue. I am noticing that the `meta` associated with a field is not moved with the `input` values. So, when you move things around with `mutators`, the input is moved, but is then associated with the meta from whatever previously existed at the new index.

This is particular problematic with Drag-and-drop. My field moves, but the `meta`, which includes things like `dirty`, `pristine`, `touched`, `valid`, etc, all stay associated with the wrong input values.

**Comment by rrjohnson85** [comment]
@311chaos sweet, I'll try it out and report back.

**Comment by erikras** [comment]
No. This is _not_ resolved. In fact, my `ffa@1.1` fix mistakenly assumed that that sort of movement was the only problem, but `unshift` is a different beast. ⚠️ Still in progress.... ⚠️

**Comment by dima19972525** [comment]
Have similar problem
